build.assets/Dockerfile-grpcbox

```diff
@@ -1,6 +1,6 @@
 # syntax=docker/dockerfile:1
 
-FROM docker.io/golang:1.21
+FROM docker.io/golang:1.22
```

Comment:
the go.mod uses `toolchain go1.22.1`, do we pin patch version here?

Reply:
We don't really care about the particular patch in the grpcbox, so I tend to not pin so I don't get yet another file I need to keep touching.

This is alright because:

1. Buf and protoc plugins are pinned
2. An older Go toolchain will silently download the newer toolchain (since 1.21 I believe)

(Just (1) would guarantee deterministic protogen though.)